lib/environ/internal/phases/restart.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+
+	"github.com/gravitational/gravity/lib/defaults"
+	libfsm "github.com/gravitational/gravity/lib/fsm"
+	"github.com/gravitational/gravity/lib/loc"
+	"github.com/gravitational/gravity/lib/storage"
+	"github.com/gravitational/gravity/lib/utils"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+// NewRestart returns a new executor to restart the runtime container to apply
+// the environment variables update
+func NewRestart(params libfsm.ExecutorParams, operator localClusterGetter, apps appGetter, operationID string, logger log.FieldLogger) (*restart, error) {
+	if params.Phase.Data == nil || params.Phase.Data.Package == nil {
+		return nil, trace.NotFound("no installed application package specified for phase %q",
+			params.Phase.ID)
+	}
+	// TODO: move these to OperationPhase.Data (i.e. to plan init phase) to avoid
+	// failing operation if a cluster is degraded
+	app, err := apps.GetApp(*params.Phase.Data.Package)
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to query installed application")
+	}
+	runtimePackage, err := app.Manifest.RuntimePackageForProfile(params.Phase.Data.Server.Role)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	cluster, err := operator.GetLocalSite()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return &restart{
+		FieldLogger:    logger,
+		runtimePackage: *runtimePackage,
+		operationID:    operationID,
+		serviceUser:    cluster.ServiceUser,
+	}, nil
+}
+
+// Execute restarts the runtime container with the new configuration package
+func (r *restart) Execute(ctx context.Context) error {
+	commands := [][]string{
+		{"system", "pull-updates",
+			"--uid", r.serviceUser.UID,
+			"--gid", r.serviceUser.GID,
+			"--runtime-package", r.runtimePackage.String(),
+			"--ops-url", defaults.GravityServiceURL,
+			"--insecure", "--quiet"},
+		{"system", "update",
```

Comment:
Yes, that's the idea - restart if either (or both) changes. Since we're rolling out a new configuration, the container needs to restart to pick it up.